### PR TITLE
Usability improvements for plan.

### PR DESCRIPTION
**Pull request by esunar**
- Added colorful input for not getting lost in logs.
- Catched invalid input values and re-request.

### CODE REVIEW

**Review by agileshaw — APPROVED** [review]
LGTM

File: cou/steps/plan.py [hunk 1]
```diff
@@ -34,18 +36,42 @@ def generate_plan(args: Namespace) -> UpgradeStep:
...
-        sys.exit(1)
-    else:
-        logging.info("Skipped")
-
-    for sub_step in upgrade_plan.sub_steps:
-        apply_plan(sub_step)
+    result = "X"
+    while result not in "CcAsSs":
+        result = input(prompt(upgrade_plan.description))
+        match result:
```
Comment (rgildein) [inline cou/steps/plan.py:64]: We will not support Python 3.8 (focal)?

File: cou/steps/plan.py [hunk 2]
```diff
@@ -34,18 +36,42 @@ def generate_plan(args: Namespace) -> UpgradeStep:
...
-        upgrade_plan.run()
-    elif result.casefold() == "a".casefold():
-        sys.exit(1)
-    else:
-        logging.info("Skipped")
-
-    for sub_step in upgrade_plan.sub_steps:
-        apply_plan(sub_step)
+    result = "X"
+    while result not in "CcAsSs":
```
Comment (rgildein) [inline cou/steps/plan.py:62]: I suggest to make this as constant + you made mistake it should be `"CcAaSs"`.
However we can do better with:
```python
while result := input(prompt(upgrade_plan.description)).lower() not in "cas":
    match result:
        case "c":
            upgrade_plan.run()
                for sub_step in upgrade_plan.sub_steps:
                    apply_plan(sub_step)
        case "a":
            logging.info("Aborning plan")
            sys.exit(1)
        case "s":
            logging.info("Skipped")
        case _:
            logging.info("No valid input provided!")
```

**Review by rgildein — CHANGES_REQUESTED** [review]

File: cou/steps/plan.py (hunk 1 above)
Comment (esunar) [reply cou/steps/plan.py:64]: baseline will be 3.10

File: cou/steps/plan.py (hunk 2 above)
Comment (esunar) [reply cou/steps/plan.py:62]: If we enter c,a or s it never runs

Comment (esunar) [reply cou/steps/plan.py]: fixed `CcAaSs` issue

Comment (esunar) [reply cou/steps/plan.py]: done

**Pull request merged**